library/Translate/test/Test/Desugar/Sexp.hs

```diff
@@ -242,3 +243,23 @@ modLetWorkAsExpected =
         \            (:let-type foo () ((XTZ))\
         \               (:record (bar) (foo))))\
         \     foo))"
+
+handlerWorkAsExpected :: T.TestTree
+handlerWorkAsExpected =
+  T.testGroup
+    "handler let desugar tests"
+    [ T.testCase
+        "defhandler -desugar-> lethandler"
+        (expected T.@=? fmap Desugar.handlerTransform form)
+    ]
+  where
+    form =
+      Sexp.parse
+        ("(:defhandler printer                     " <>
+         "  ((:defop print () printLn)             " <>
+         "   (:defop pure (x) (toString x))))      ")
+    expected =
+      Sexp.parse
+        ("(:lethandler printer                     " <>
+         "  (:ops (:defop print () printLn))       " <>
+         "  (:defret (x) (toString x)))            ")
```

Comment:
What is this `:let-handler` is over? At least `let` typically implies being let over something. If not might be good to change the name to not confuse it with all the other `let` forms.